recipes/cmake/3.x.x/conanfile.py

```diff
@@ -68,73 +70,73 @@ def validate(self):
+            destination=self.source_folder, strip_root=True)
+        rmdir(self, os.path.join(self.source_folder, "Tests", "RunCMake", "find_package"))
 
-    def _configure_cmake(self):
-        if not self._cmake:
-            self._cmake = CMake(self)
+    def generate(self):
+        if self.options.bootstrap:
+            tc = AutotoolsToolchain(self)
+            tc.generate()
+            bootstrap_cmake_options = ["--"]
+            bootstrap_cmake_options.append(f'-DCMAKE_CXX_STANDARD={"11" if not self.settings.compiler.cppstd else self.settings.compiler.cppstd}')
+            if self.settings.os == "Linux":
+                if self.options.with_openssl:
+                    openssl = self.dependencies["openssl"]
+                    bootstrap_cmake_options.append("-DCMAKE_USE_OPENSSL=ON")
+                    bootstrap_cmake_options.append(f'-DOPENSSL_USE_STATIC_LIBS={"FALSE" if openssl.options.shared else "TRUE"}')
+                else:
+                    bootstrap_cmake_options.append("-DCMAKE_USE_OPENSSL=OFF")
+            files.save_toolchain_args({"bootstrap_cmake_options": ' '.join(arg for arg in bootstrap_cmake_options)}, namespace="bootstrap")
+        else:
+            tc = CMakeToolchain(self)
             if not self.settings.compiler.cppstd:
-                self._cmake.definitions["CMAKE_CXX_STANDARD"] = 11
-            self._cmake.definitions["CMAKE_BOOTSTRAP"] = False
+                tc.variables["CMAKE_CXX_STANDARD"] = 11
+            tc.variables["CMAKE_BOOTSTRAP"] = False
             if self.settings.os == "Linux":
-                self._cmake.definitions["CMAKE_USE_OPENSSL"] = self.options.with_openssl
+                tc.variables["CMAKE_USE_OPENSSL"] = self.options.with_openssl
                 if self.options.with_openssl:
-                    self._cmake.definitions["OPENSSL_USE_STATIC_LIBS"] = not self.options["openssl"].shared
-            if tools.cross_building(self):
-                self._cmake.definitions["HAVE_POLL_FINE_EXITCODE"] = ''
-                self._cmake.definitions["HAVE_POLL_FINE_EXITCODE__TRYRUN_OUTPUT"] = ''
-            self._cmake.configure(source_folder=self._source_subfolder)
-
-        return self._cmake
+                    openssl = self.dependencies["openssl"]
+                    tc.variables["OPENSSL_USE_STATIC_LIBS"] = not openssl.options.shared
+            if cross_building(self):
+                tc.variables["HAVE_POLL_FINE_EXITCODE"] = ''
+                tc.variables["HAVE_POLL_FINE_EXITCODE__TRYRUN_OUTPUT"] = ''
+            tc.generate()
 
     def build(self):
         if self.options.bootstrap:
-            with tools.chdir(self._source_subfolder):
-                self.run(['./bootstrap', '--prefix={}'.format(self.package_folder), '--parallel={}'.format(tools.cpu_count())])
-                autotools = AutoToolsBuildEnvironment(self)
+            toolchain_file_content = files.load_toolchain_args(self.generators_folder, namespace="bootstrap")
```

Comment:
not allowed, it's not part of conan public API and may break in future release

Reply:
Replaced with json.loads(load()).